### PR TITLE
Fix 10.0.x migration guide

### PR DESCRIPTION
**Pull request by jonas**
Change `AkkaHttp` to `HttpApp` in the migration guide. I also took the liberty to rename the file to better match the others, the title remains the same.

### CODE REVIEW

File: docs/src/main/paradox/scala/http/migration-guide/index.md [hunk 1]
```diff
@@ -7,6 +7,6 @@
 
 * [migration-from-spray](migration-from-spray.md)
 * [migration-guide-2.4.x-10.0.x](migration-guide-2.4.x-10.0.x.md)
-* [migration-within-10.0.x-Api-May-Change](migration-within-10.0.x-Api-May-Change.md)
+* [migration-guide-10.0.x](migration-guide-10.0.x.md)
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/migration-guide/index.md]: No strong feelings, now it matches the pattern. I used this name to show clearly that it's only about changes on `ApiMayChange`.

Comment (jonas) [inline docs/src/main/paradox/scala/http/migration-guide/index.md]: My main worry is that at some point we will switch to Akka 2.5 and we'll need to document some hoops to jump through when moving from Akka 2.4, like migrating to use AbstractActor.

Comment (jlprat) [inline docs/src/main/paradox/scala/http/migration-guide/index.md]: Fair point.

Comment (jrudolph) [inline docs/src/main/paradox/scala/http/migration-guide/index.md]: Yes, let's keep it simple and collect all those information in one place.

In the middle-term, we'll try not to lose compatibility with 2.4 so it should at least be possible to still use Akka 2.4 for a while regardless of whether we compile our binaries against 2.5 or 2.4.

**Pull request merged**